odl/set/domain.py

```diff
@@ -41,21 +41,23 @@ class IntervalProd(Set):
 
     An interval product is a Cartesian product of n intervals, i.e. an
     n-dimensional rectangular box aligned with the coordinate axes
-    as a subset of :math:`R^n`.
+    as a subset of the n-dimensional Euclidean space.
 
-    `IntervalProd` objects are immutable, all methods involving them return
-    a new `IntervalProd`.
+    `IntervalProd` objects are immutable, hence all manipulation methods
+    return a new instance.
     """
 
     def __init__(self, begin, end):
         """Initialize a new instance.
 
         Parameters
         ----------
-        begin : `array-like` or `float`
-            The lower ends of the intervals in the product
-        end : `array-like` or `float`
-            The upper ends of the intervals in the product
+        begin : `array-like` or float
```

Comment:
```
begin, end : float or `array-like`
```

I.m.o. put the scalar as the first option (being the most simple) and you can condense this to a single entry with no loss of information.